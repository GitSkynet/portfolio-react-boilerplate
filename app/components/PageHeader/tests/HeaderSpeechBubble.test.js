import React from 'react';
import { shallow, mount } from 'enzyme';

import HeaderSpeechBubble from '../HeaderSpeechBubble';

describe('<HeaderSpeechBubble />', () => {
  it('Should render and match the snapshot', () => {
    const tree = shallow(<HeaderSpeechBubble />);
    expect(tree).toMatchSnapshot();
  });

  it('Should adopt the "isLeftHanded" prop', () => {
    const tree = mount(<HeaderSpeechBubble isLeftHanded={false} />);
    expect(tree.prop('isLeftHanded')).toBe(false);
  });
});
