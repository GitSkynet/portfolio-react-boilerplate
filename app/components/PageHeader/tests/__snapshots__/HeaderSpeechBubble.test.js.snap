// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HeaderSpeechBubble /> Should render and match the snapshot 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div
    className="HeaderSpeechBubble__HeaderSpeechBubble-jNrgAS iBRuwE"
/>,
  "nodes": Array [
    <div
      className="HeaderSpeechBubble__HeaderSpeechBubble-jNrgAS iBRuwE"
/>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <HeaderSpeechBubble__HeaderSpeechBubble
        isLeftHanded={true}
/>,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": StyledComponent {
        "_reactInternalInstance": [Circular],
        "attrs": Object {},
        "context": Object {
          "__styled-components-stylesheet__": undefined,
          "__styled-components__": undefined,
        },
        "props": Object {
          "isLeftHanded": true,
        },
        "refs": Object {},
        "state": Object {
          "generatedClassName": "iBRuwE",
          "theme": Object {},
        },
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div
          className="HeaderSpeechBubble__HeaderSpeechBubble-jNrgAS iBRuwE"
/>,
        "_debugID": 2,
        "_renderedOutput": <div
          className="HeaderSpeechBubble__HeaderSpeechBubble-jNrgAS iBRuwE"
/>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <HeaderSpeechBubble__HeaderSpeechBubble
    isLeftHanded={true}
/>,
}
`;
